### PR TITLE
Add uvdata

### PR DESCRIPTION
**Pull request by adampbeardsley**
This branch adds the ability to combine uvdata objects along baseline-time, polarization, or frequency axes (or combinations). It also allows for in place adding (`uv1 += uv2`), and allows for multiple file reads (which really just iterates through the list, adding objects as it goes).

### CODE REVIEW

File: pyuvdata/tests/test_uvdata.py [hunk 1]
```diff
@@ -661,3 +661,172 @@ def test_reorder_pols():
...
+    times = np.unique(uv_full.time_array)
+    uv1.select(times=times[0:len(times) / 2])
+    uv2.select(times=times[len(times) / 2:])
+    uv1 += uv2
+    nt.assert_equal(uv_full.history + '  Downselected to specific times'
+                    ' using pyuvdata. Combined data along baseline-time axis using'
+                    ' pyuvdata.', uv1.history)
+    uv1.history = uv_full.history
+    nt.assert_equal(uv1, uv_full)
+
```
Comment (adampbeardsley) [inline pyuvdata/tests/test_uvdata.py]: I treat the baseline-time axis as a single entity to add. In the testing, it was easiest to trim/add by times. I'll add another one that does it by baseline for completeness.

**Comment by adampbeardsley** [comment]
Do I need to do anything to update the docs with new docstrings?

**Comment by bhazelton** [comment]
No, the docstrings are automatically included in the docs. The only time we need to update docs is if we modify the UVParameters in any way (change descriptions, add/remove them etc) or to update the tutorial.

**Comment by bhazelton** [comment]
Looks great! I'm merging.

**Pull request merged**